6_visualize/src/prep_timeline_fun.R

```diff
@@ -0,0 +1,59 @@
+prep_timeline_fun <- function(DateTime, timestep_ind, timeline_config, spark_config, legend_text_cfg) {
+
+  this_DateTime <- as.POSIXct(DateTime, tz = "UTC") # WARNING, IF WE EVER MOVE FROM UTC elsewhere, this will be fragile/bad.
+
+  # Compute the full x limits for all datetimes
+  storm_timesteps <- fetch_read(timestep_ind)
+  date_lims <- as.POSIXct(range(storm_timesteps), tz = "UTC")
+
+  # Define functions that will convert each site's shapes to user coordinates
+  dateTime_to_x <- function(dateTime, x_user) {
+    date_lims_num <- as.numeric(date_lims, units='days')
+    date_time_num <- as.numeric(dateTime, units='days')
+    x_frac <- (date_time_num - date_lims_num[1]) / diff(date_lims_num)
+    x_user[1] + x_frac*diff(x_user)
+  }
+
+  # Create and return a closure/function that can be called to add sparklines
+  # when making the full plot
+  plot_fun <- function(){
+
+    # Ask the open device for the user coordinates
+    coord_space <- par()$usr
+
+    # Convert normalized plot coordinates to user coordinates
+    loc_x <- coord_space[1] + c(spark_config$xleft, spark_config$xright) * diff(coord_space[1:2])
+    loc_y <- coord_space[3] + spark_config$ytimeline * diff(coord_space[3:4])
+    loc_date_y <- coord_space[3] + spark_config$ytimeline_text * diff(coord_space[3:4])
+
+    # Convert this the current date & start/end to user coordinates
+    loc_date_x <- dateTime_to_x(this_DateTime, loc_x)
+    loc_start_x <- dateTime_to_x(date_lims[1], loc_x)
+    loc_end_x <- dateTime_to_x(date_lims[2], loc_x)
+
+    # Add timeline
+    points(loc_x, rep(loc_y, length(loc_x)), type = 'l', col = timeline_config$line_col,
+           lwd = timeline_config$line_lwd, lty = timeline_config$line_lty)
```

Comment:
I think a tiny vertical tick for the marker may be a better choice than the triangle, since the triangle adds another "marker" (e.g., dots, circles) to the mix and you can avoid that w/ using a small tick instead? just a thought. I think you were looking at different options here.

Reply:
The vertical bar feels a bit dated to me. I sketched out a few different ideas [here in Teams](https://teams.microsoft.com/l/message/19:79eaaf0209574972b1afd7c5453fc11e@thread.skype/1593439144163?tenantId=0693b5ba-4b18-4d7b-9341-f32f400a5494&groupId=80983861-ab36-4fdc-a217-17742ba5ee3d&parentMessageId=1593188119326&teamName=GS-WMA-IIDD-Makerspace&channelName=General&createdTime=1593439144163) which is how the triangle idea came to be. 

Here's what it looks like with a bar:

![animation_a](https://user-images.githubusercontent.com/13220910/86163759-b75c3180-bade-11ea-84d5-7327f5ad1249.gif)

Reply:
yeah, that looks dated :). Was thinking more like this:
![image](https://user-images.githubusercontent.com/2349007/86239725-7a874d80-bb65-11ea-9109-c786420e6d4c.png) (or even smaller)

Reply:
OK, so that is definitely better than the tick version I implemented. I am still on the fence though. I like the simplicity but I'm struggling to think about how modern or not modern it is. Probably better than an arrow, though.

I'm not sure I would want it even smaller because then it might not be seen at all.

Reply:
Most recent commit results in:

![animation_a](https://user-images.githubusercontent.com/13220910/86276424-d623fc00-bba2-11ea-8b06-41072b998725.gif)

Reply:
what do you think after seeing it? Personally, I prefer it because it is simple and doesn't introduce a new marker type, but I wouldn't shed tears over you going in the other direction :)

Reply:
I do like the simplicity and the kind of "wiping" effect it creates with moving at the same time as the spark lines. I also liked the triangle but don't feel like it makes anything that much better so I am good with going for simple!

Reply:
:shipit: